traffic_ops/testing/api/v14/api_capability_test.go

```diff
@@ -0,0 +1,77 @@
+package v14
+
+/*
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import (
+	"testing"
+)
+
+func TestAPICapabilities(t *testing.T) {
+	testCases := []struct {
+		description string
+		capability  string
+		order       string
+		first       string
+		count       int
+	}{
+		{
+			description: "Successfully get all asns-write API Capabilities",
+			capability:  "asns-write",
+			count:       3,
+		},
+		{
+			description: "Successfully get all asns-read API Capabilities",
+			capability:  "asns-read",
+			count:       2,
+		},
+		{
+			description: "Successfully get all cache-groups-read API Capabilities",
+			capability:  "cache-groups-read",
+			count:       9,
+		},
+		{
+			description: "Fail to get any API Capabilities with a bogus capability",
+			capability:  "foo",
+			count:       0,
+		},
+		{
+			description: "Successfully get all API Capabilities in order of HTTP Method",
+			order:       "httpMethod",
+			first:       "GET",
+		},
+	}
+
+	for _, c := range testCases {
+		t.Run(c.description, func(t *testing.T) {
+			caps, _, err := TOSession.GetAPICapabilities(c.capability, c.order)
+
+			if err != nil {
+				t.Fatalf("error: expected error result: %v", err)
+			}
+
+			if c.count != 0 && len(caps.Response) != c.count {
```

Comment:
Is it necessary to check `c.count != 0`? This may never actually happen, but it *is* possible to have a Capability without any routes that require it. Plus above you're specifying explicitly that you expect 0 returned results for capability "foo" but that's never checked. Maybe instead you should check for `c.count >= 0` and in cases where you don't care set it to `-1` explicitly?